File: backend/src/main/java/com/drip/admin/modules/system/controller/MenuController.java
package com.drip.admin.modules.system.controller;

import com.drip.admin.common.log.OperationLog;
import com.drip.admin.common.response.ApiResponse;
import com.drip.admin.common.security.RequirePermission;
import com.drip.admin.modules.system.dto.*;
import com.drip.admin.modules.system.service.MenuService;
import com.drip.admin.modules.system.vo.MenuTreeVo;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.*;

import static com.drip.admin.shared.utils.AdminUtils.*;


@RestController
@RequestMapping("/system")
public class MenuController {
    private final MenuService menuService;

   public MenuController(MenuService menuService) {
        this.menuService = menuService;
    }

    @GetMapping("/menu")
    @RequirePermission("system:menu:list")
    public ApiResponse<List<MenuTreeVo>> menus() {
        return ApiResponse.success(menuService.tree());
    }

    @PostMapping("/menu")
    @RequirePermission("system:menu:create")
    @OperationLog(module = "菜单管理", action = "新增菜单")
    public ApiResponse<Long> createMenu(@Valid @RequestBody MenuSaveRequest request) {
        return ApiResponse.success(menuService.create(request));
    }

    @PutMapping("/menu/{id}")
    @RequirePermission("system:menu:update")
    @OperationLog(module = "菜单管理", action = "编辑菜单")
    public ApiResponse<Void> updateMenu(@PathVariable long id, @Valid @RequestBody MenuSaveRequest request) {
        menuService.update(id, request);
        return ApiResponse.success(null);
    }

    @DeleteMapping("/menu/{id}")
    @RequirePermission("system:menu:delete")
    @OperationLog(module = "菜单管理", action = "删除菜单")
    public ApiResponse<Void> deleteMenu(@PathVariable long id) {
        menuService.delete(id);
        return ApiResponse.success(null);
    }

    @PutMapping("/menu/{id}/status")
    @RequirePermission("system:menu:status")
    @OperationLog(module = "菜单管理", action = "变更菜单状态")
    public ApiResponse<Void> menuStatus(@PathVariable long id, @Valid @RequestBody StatusUpdateRequest request) {
        menuService.updateStatus(id, request.statusOrDefault());
        return ApiResponse.success(null);
    }
}

File: backend-go/internal/modules/system/controller/menu_controller.go
package controller

import (
	"drip-admin/backend-go/internal/common"
	"drip-admin/backend-go/internal/modules/system/dto"
	"github.com/gin-gonic/gin"
)

var (
	_ = common.ApiResponse{}
	_ = dto.MenuSaveRequest{}
)

// Menus returns the menu tree.
// @Summary List menus
// @Tags menu
// @Produce json
// @Security ApiKeyAuth
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/menu [get]
func (ctl *Controller) Menus(c *gin.Context) {
	ctl.svc.Menus(c)
}

// CreateMenu creates a menu.
// @Summary Create menu
// @Tags menu
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param request body dto.MenuSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/menu [post]
func (ctl *Controller) CreateMenu(c *gin.Context) {
	ctl.svc.CreateMenu(c)
}

// UpdateMenu updates a menu.
// @Summary Update menu
// @Tags menu
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Menu ID"
// @Param request body dto.MenuSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/menu/{id} [put]
func (ctl *Controller) UpdateMenu(c *gin.Context) {
	ctl.svc.UpdateMenu(c)
}

// DeleteMenu deletes a menu.
// @Summary Delete menu
// @Tags menu
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Menu ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 409 {object} common.ApiResponse
// @Router /system/menu/{id} [delete]
func (ctl *Controller) DeleteMenu(c *gin.Context) {
	ctl.svc.DeleteMenu(c)
}

// MenuStatus updates a menu status.
// @Summary Update menu status
// @Tags menu
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Menu ID"
// @Param request body dto.StatusUpdateRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/menu/{id}/status [put]
func (ctl *Controller) MenuStatus(c *gin.Context) {
	ctl.svc.MenuStatus(c)
}
